dessia_common/graph.py

```diff
@@ -56,3 +56,43 @@ def extract_region(networkx_graph: nx.Graph, nodes, distance: int = 5):
         region_nodes.extend(ancs)
 
     return networkx_graph.subgraph(region_nodes)
+
+
+def get_longest_path(graph: nx.DiGraph, begin, end) -> list:
+    return list(nx.shortest_simple_paths(graph, begin, end))[-1]
+
+
+def get_paths_from_to(graph: nx.DiGraph, _from: list, _to: list) -> list:
+    paths = []
+    for origin in _from:
+        for destination in _to:
+            try:
+                paths.append(get_longest_path(graph, origin, destination))
+            except nx.exception.NetworkXNoPath:
+                continue
+    return paths
+
+
+def get_distance_by_nodes(graph: nx.DiGraph) -> Dict:
+    longest_path = nx.dag_longest_path(graph)
+    end_of_path = longest_path[-1]
+
+    distances = {}
+
+    for node in longest_path:
+        distances[node] = longest_path.index(node)
+
+    for node in [n for n in graph.nodes if n not in longest_path]:
+        try:
+            path = get_longest_path(graph, node, end_of_path)
+            distances[node] = len(longest_path) - len(path)
+        except nx.exception.NetworkXNoPath:
+            pass
```

Comment:
What means this exception and why are we avoiding it ?

Reply:
This exception is raised when there is no path between `node` and `end_of_path`. 
In such a case, we look for a path from any treated node to `node`

In fact, we could write this, but I'm not sure that this is more readable: 
```py 
    for node in [n for n in graph.nodes if n not in longest_path]:
        try:
            path = get_longest_path(graph, node, end_of_path)
            distances[node] = len(longest_path) - len(path)
        except nx.exception.NetworkXNoPath:
            for path in get_paths_from_to(graph, list(graph.nodes), [node]):
                distance_via_current_node = distances[path[0]] + len(path) - 1
                if node not in distances or distances[node] < distance_via_current_node:
                    distances[node] = distance_via_current_node
```